### ISSUE TITLE
[[BUILD-version 2.7.2]] --enable-iscsi fail

### ISSUE THREAD

**Issue by Obarun** [storaged-project/udisks#388]
hi,
trying to build the last version build with --enable-iscsi modules option.

> checking libiscsi.h usability... no
checking libiscsi.h presence... no
checking for libiscsi.h... no
checking for libiscsi_init in -liscsi... no
configure: error: iSCSI support requested but libraries not found

it look after libiscsi.h, i think this is should be iscsi.h, not?

**Comment by vojtechtrefny** [comment]
No, it should really be `libiscsi.h`. It is provided by `iscsi-initiator-utils-devel` package in Fedora. I don't know if this is packaged in ArchLinux.

**Comment by Obarun** [comment]
in this case it's not possible to build it correctly on Archlinux. Archlinux provide the package libiscsi coming from this upstream : https://github.com/sahlberg/libiscsi. According to this upstream source, the h source is called iscsi.h ( https://github.com/sahlberg/libiscsi/tree/master/include ) if i do not make a mistake.
May be your upstream source for libiscsi is not the same? In this case, can you give me your uptsream source about libiscsi?

**Comment by vojtechtrefny** [comment]
Yes, this is actually a different "libiscsi". I wasn't able to find a better "upstream" source for "our" libiscsi than this --https://src.fedoraproject.org/rpms/iscsi-initiator-utils/blob/master/f/0009-libiscsi.patch -- patch for _iscsi-initiator-utils_ package.

I don't know why udisks uses this libsicsi and not the one you found.

**Comment by Obarun** [comment]
ok, got the trouble, udisks use open-iscsi https://github.com/open-iscsi/open-iscsi instead of libiscsi https://github.com/sahlberg/libiscsi, changing my packages dependencies should solve the trouble. i will test it.
Thanks for your help and time

**Comment by Obarun** [comment]
So i fact, upstream source of open-iscsi do not provide libiscsi.h helper file. Fedora create this file by patching the source. That's mean that you provide a modules which it only works on Fedora system.

**Comment by vojtechtrefny** [comment]
Yes, this is not ideal situation. I'm not author of the plugin and I don't know why they decided to use this library and not something that is upstreamed. But there is actually some good news -- the library should be soon part of openiscsi upstream -- some parts are already merged (https://github.com/open-iscsi/open-iscsi/pull/71). For now you can just compile udisks without the iscsi plugin.

(I'm sorry it took me so long to answer. I completely forgot about this issue.)

**Comment by Obarun** [comment]
no problem :), this is happen sometime

**Comment by greyltc** [comment]
Is there any news here? Can the fedora patch dependency be dropped in favor of a dependency on open-iscsi for the `--enable-iscsi` build option?

**Comment by tbzatek** [comment]
No actual work has been done on this. The upstream `libopeniscsiusr` library needs to be enhanced with missing operations and the UDisks plugin needs to be rewritten. It is on my TODO list though.

**Comment by dvzrv** [comment]
Hi! :wave: 

I package udisks2 for Arch Linux and would be happy to see this being done, so that we may also eventually ship the iSCSI plugin.

I'll subscribe this ticket for further news on the topic :sweat_smile:

**Comment by tbzatek** [comment]
Thanks for your interest. The status is still "No actual work has been done on this."
